### PR TITLE
AF-362 zatfile

### PR DESCRIPTION
**Pull request by sandlerr**
@zendesk/vegemite solves https://support.zendesk.com/hc/en-us/community/posts/211654267-suggestion-provide-a-zatfile-parameter-to-zat

https://zendesk.atlassian.net/browse/AF-362

### CODE REVIEW

**Comment by ocke** [comment]
Can I suggest another solution that is much more flexible and holistic?

a `.zat` file in your home directory ~/.zat that is a yaml file:

```
support:
  domain: support.zd-dev.com
  username: admin@zendesk.com
  token: 12345

z3nzibar:
  username: okwant@zendesk.com
  token: 12345

z3nzibarS:
  domain: z3nzibar1234543.zendesk.com
  username: okwant@zendesk.com
  token: 12345
```

If you leave out `domain` it will use name + `.zendesk.com`.

This way users can upload any app to any domain with 1 settings file. Also including Sandboxes.

**Comment by sandlerr** [comment]
I like it! We'll still need multiple .zat files as each app has its own ID, so we'll need to create `.zat.z3nzibar` when you do `zat create --domain=z3nzibar` right?

What if I install `myapp-test` and `myapp-prod` under the same subdomain?

**Comment by sandlerr** [comment]
Also, anyone know offhand if we can use `~` in pathnames on windows?

File: doc/tools.md [hunk 1]
```diff
@@ -112,3 +114,14 @@ Removes the zip files in the <tt>tmp</tt> folder that's created when you package
 
     $ zat clean
 
+### Create
+
+Packages your app directory into a zipfile, then uploads it into a Zendesk account. Also stores the new app's ID and other metadata in a zatfile (by default, this is `.zat`).
```
Comment (chucknado) [inline doc/tools.md]: zipfile -> zip file (2 words)
zatfile -> zat file (2 words if not referring to the command-line option `zatfile`)

File: doc/tools.md [hunk 2]
```diff
@@ -112,3 +114,14 @@ Removes the zip files in the <tt>tmp</tt> folder that's created when you package
...
+
+A `path` option pointing to the directory containing the app can be specified on the command-line using `zat create --path=./MY_PATH`. The name of the `zatfile` can be passed using `zat create --zatfile=.zat.test`.
```
Comment (chucknado) [inline doc/tools.md]: Original:
> A `path` option pointing to the directory containing the app can be specified on the command-line using `zat create --path=./MY_PATH`. The name of the `zatfile` can be passed using `zat create --zatfile=.zat.test`.

Sentence structured as "can be this" and "can be that", which is okay but not optimal in terms of the tasks the dev is trying to accomplish ("I want to this" and "I want to do that"). Suggested edit to make it a little more specific:

> You can point to the directory containing the app by using the `path` option. Example: `zat create --path=./MY_PATH`. You can specify the name of the zat file with the `zatfile` option. Example: `zat create --zatfile=.zat.test`.

* command-line -> command line (no hyphen if used as a noun)

* "the name of the `zatfile" -> "the name of the zat file" -> 2 words and no code-case if referring to a file, not the option

File: doc/tools.md [hunk 3]
```diff
@@ -112,3 +114,14 @@ Removes the zip files in the <tt>tmp</tt> folder that's created when you package
...
+
+An existing zipfile may be used instead of an app directory by passing a `zipfile` option like `zat create --zipfile=~/zendesk_app/my_zipfile.zip`.
+
+### Update
+
+Much like create, this command is used to update an app that you have previously create using `zat create`. This command updates the existing app using the app ID and other metadata found in the `zatfile`. By default this file is called `.zat` in the app's directory.
```
Comment (chucknado) [inline doc/tools.md]: Original: 
> Much like create, this command is used to update an app ...

Suggested edit to switch to the active voice:
> Much like create, use this command to update an app ...

File: doc/tools.md [hunk 4]
```diff
@@ -112,3 +114,14 @@ Removes the zip files in the <tt>tmp</tt> folder that's created when you package
 
     $ zat clean
 
+### Create
+
+Packages your app directory into a zipfile, then uploads it into a Zendesk account. Also stores the new app's ID and other metadata in a zatfile (by default, this is `.zat`).
+
+A `path` option pointing to the directory containing the app can be specified on the command-line using `zat create --path=./MY_PATH`. The name of the `zatfile` can be passed using `zat create --zatfile=.zat.test`.
+
+An existing zipfile may be used instead of an app directory by passing a `zipfile` option like `zat create --zipfile=~/zendesk_app/my_zipfile.zip`.
```
Comment (chucknado) [inline doc/tools.md]: zipfile -> zip file

**Comment by adammw** [comment]
:+1:

**Comment by sandlerr** [comment]
@ocke @zendesk/vegemite I've re-written this PR to do as Olaf suggested (although using JSON rather than YAML for consistency with `.zat`).

**Comment by ocke** [comment]
:+1:

**Pull request merged**